src/libNetwork/P2PComm.cpp

```diff
@@ -583,6 +606,81 @@ void P2PComm::EventCallback(struct bufferevent* bev, short events,
         // Queue the message
         m_dispatcher(raw_message);
     }
+    else if (startByte == START_BYTE_GOSSIP)
+    {
+        // Check for length consistency
+
+        if (messageLength != message.size() - HDR_LEN)
+        {
+            LOG_GENERAL(WARNING, "Incorrect message length.");
+            return;
+        }
+
+        if (messageLength < GOSSIP_MSGTYPE_LEN + GOSSIP_ROUND_LEN
+                + GOSSIP_SNDR_LISTNR_PORT_LEN)
+        {
+            LOG_GENERAL(
+                WARNING,
+                "Gossip Msg Type and/or Gossip Round and/or SNDR LISTNR "
+                "Port is missing (messageLength = "
+                    << messageLength << ")");
+            return;
+        }
+
+        std::vector<unsigned char> gossipMsgTyp(message.begin() + HDR_LEN,
```

Comment:
It seems like we could have just used `unsigned char gossipMsgType = message.at(HDR_LEN);`

Reply:
True.